mutiny-core/src/logging.rs

```diff
@@ -135,93 +123,24 @@ impl Logger for MutinyLogger {
-        .add_object_store(ObjectStore::new(LOGGING_OBJECT_STORE_NAME))
-        .build()
-        .await?;
-
-    Ok(rexie)
-}
-
-#[cfg(any(test, feature = "test-utils"))]
-pub(crate) async fn clear(db_prefix: Option<String>) -> Result<(), MutinyError> {
-    let indexed_db = build_logging_database(db_prefix).await?;
-    let tx = indexed_db.transaction(&[LOGGING_OBJECT_STORE_NAME], TransactionMode::ReadWrite)?;
-    let store = tx.store(LOGGING_OBJECT_STORE_NAME)?;
-
-    store.clear().await?;
-
-    tx.done().await?;
-
-    Ok(())
-}
-
-async fn get_logging_data(rexie: &Rexie) -> Result<Option<Vec<String>>, MutinyError> {
-    // Create a new read-only transaction
-    let transaction = rexie.transaction(&[LOGGING_OBJECT_STORE_NAME], TransactionMode::ReadOnly)?;
-
-    let mut store = transaction.store(LOGGING_OBJECT_STORE_NAME)?;
-
-    let res = get_logging_data_with_transaction_store(&mut store).await;
-
-    transaction.done().await?;
-
-    res
-}
-
-/// Pass in a transaction to get the data for.
-/// It is up to the caller to call `transaction.done()`
-async fn get_logging_data_with_transaction_store(
-    store: &mut rexie::Store,
-) -> Result<Option<Vec<String>>, MutinyError> {
-    let logging_js = store.get(&JsValue::from(LOGGING_KEY)).await?;
-
-    // If the key doesn't exist, we return None
-    if logging_js.is_null() || logging_js.is_undefined() {
-        return Ok(None);
-    }
-
-    let logging_data: Vec<String> = logging_js.into_serde()?;
-
-    Ok(Some(logging_data))
+fn get_logging_data<S: MutinyStorage>(storage: &S) -> Result<Option<Vec<String>>, MutinyError> {
+    storage.get_data(LOGGING_KEY)
 }
 
-async fn write_logging_data(
-    rexie: &Rexie,
-    recent_logs: &mut Vec<String>,
+async fn write_logging_data<S: MutinyStorage>(
+    storage: &S,
+    mut recent_logs: Vec<String>,
 ) -> Result<(), MutinyError> {
-    // Create a new read-write transaction
-    let transaction =
-        rexie.transaction(&[LOGGING_OBJECT_STORE_NAME], TransactionMode::ReadWrite)?;
-
-    let mut store = transaction.store(LOGGING_OBJECT_STORE_NAME)?;
-
     // get the existing data so we can append to it, trimming if needed
-    let mut existing_logs = get_logging_data_with_transaction_store(&mut store)
-        .await?
-        .unwrap_or_default();
-    existing_logs.append(recent_logs);
+    let mut existing_logs: Vec<String> = get_logging_data(storage)?.unwrap_or_default();
+    existing_logs.append(&mut recent_logs);
     if existing_logs.len() > MAX_LOG_ITEMS {
         let start_index = existing_logs.len() - MAX_LOG_ITEMS;
         existing_logs.drain(..start_index);
     }
 
     // Save the logs
-    store
-        .put(
-            &JsValue::from_serde(&serde_json::to_value(existing_logs).unwrap()).unwrap(),
-            Some(&JsValue::from(LOGGING_KEY)),
-        )
-        .await?;
-
-    // Waits for the transaction to complete
-    transaction.done().await?;
+    storage.set_data(LOGGING_KEY, &existing_logs)?;
```

Comment:
We no longer have access to the rexie internals anymore because of the storage trait, so we can't really get that guarantee. it is just logs so don't think it is the end of the world.